### PR TITLE
Fix off by one error in RelTime

### PR DESCRIPTION
**Pull request by dustin**
This change fixes a user reported issue where the `RelTime` formatter picks the wrong magnitude at the exact nanosecond it changes.

User report:

```
package main

import (
	"fmt"
	"time"

	"google3/base/go/google"
	"google3/third_party/golang/humanize/humanize"
)

func main() {
	google.Init()
	fmt.Println(humanize.RelTime(time.Unix(0, 0), time.Unix(7*24*60*60, 0), "ago", ""))
	fmt.Println(humanize.RelTime(time.Unix(0, 0), time.Unix(7*24*60*60, 1), "ago", ""))
	fmt.Println(humanize.RelTime(time.Unix(0, 0), time.Unix(14*24*60*60, 0), "ago", ""))
	fmt.Println(humanize.RelTime(time.Unix(0, 0), time.Unix(14*24*60*60, 1), "ago", ""))
}
```

produces the following output:

```
7 days ago
1 week ago
1 week ago
2 weeks ago
```

### CODE REVIEW

**Comment by dustin** [comment]
Travis reports `t.Helper` fails on a bunch of the builders.  I've removed that change from this PR.

File: times_test.go [hunk 1]
```diff
@@ -37,6 +37,15 @@ func TestPast(t *testing.T) {
 	}.validate(t)
 }
 
+func TestReltimeOffbyone(t *testing.T) {
+	testList{
+		{"a", RelTime(time.Unix(0, 0), time.Unix(7*24*60*60, 0), "ago", ""), "1 week ago"},
+		{"b", RelTime(time.Unix(0, 0), time.Unix(7*24*60*60, 1), "ago", ""), "1 week ago"},
+		{"c", RelTime(time.Unix(0, 0), time.Unix(14*24*60*60, 0), "ago", ""), "2 weeks ago"},
+		{"d", RelTime(time.Unix(0, 0), time.Unix(14*24*60*60, 1), "ago", ""), "2 weeks ago"},
```
Comment (shurcooL) [inline times_test.go]: Can you add two more test cases, `time.Unix(7*24*60*60, -1)` and `time.Unix(14*24*60*60, -1)`, for complete coverage of the boundary?

**Comment by dustin** [comment]
Added some tests and renamed these conditions.  Does this behavior seem reasonable?

**Comment by dustin** [comment]
(accidentally pushed to head -- fixed it)

File: times_test.go (hunk 1 above)
Comment (dustin) [reply times_test.go]: Note that it says  "6 days" one nanosecond before a week.

File: times_test.go [hunk 2]
```diff
@@ -37,6 +37,17 @@ func TestPast(t *testing.T) {
 	}.validate(t)
 }
 
+func TestReltimeOffbyone(t *testing.T) {
+	testList{
+		{"1w-1", RelTime(time.Unix(0, 0), time.Unix(7*24*60*60, -1), "ago", ""), "6 days ago"},
+		{"1w", RelTime(time.Unix(0, 0), time.Unix(7*24*60*60, 0), "ago", ""), "1 week ago"},
```
[the remark this answers is not in the document]
Comment (dustin) [reply times_test.go]: Heh, I've been using the go font for go code, so I don't remember what alignment looks like.  :)  Added, though.

[the remark this answers is not in the document]
Comment (dustin) [reply times_test.go]: Apologies for missing that comment.  github's code review is a little foreign to me.  I'm used to the ones where you have to actually take action when someone calls something out (that is, if I ignore it, I have to at least do it intentionally).  It does seem easy to drop stuff here.

Not sure about the tests, though.  I'd've hoped a lot would break changing a `>=` to a `>`

**Comment by shurcooL** [comment]
> Heh, I've been using the go font for go code, so I don't remember what alignment looks like. :)

Me too! Wait, do you mean you use the actual "Go" font, the non-monospaced one? I use "Go Mono".

> Apologies for missing that comment. github's code review is a little foreign to me. I'm used to the ones where you have to actually take action when someone calls something out (that is, if I ignore it, I have to at least do it intentionally). It does seem easy to drop stuff here.

Makes sense; no worries.

> Not sure about the tests, though.  I'd've hoped a lot would break changing a `>=` to a `>`

I'm guessing it's related to that PR #26 being merged as df564ff1a922025fc8284d8894b701033dbff46b with some additional changes from your similar PR #33 merged as 3143592b5ed17c9246000db220863a09f86c1970 on top. Oh well, I'm not too worried about it by now.

**Comment by dustin** [comment]
Yeah, I started using the non-mono one.  We don't have hard line length limitations at work, and go doesn't do very fancy alignment, so I gave it a shot to see how I'd like it.  It's not that bad, but it was a pretty big leap to program without a fixed width font.  I still used a fixed with font for everything else.

**Comment by shurcooL** [comment]
> > Not sure about the tests, though.  I'd've hoped a lot would break changing a `>=` to a `>`
>
> I'm guessing it's related to that PR #26 being merged as df564ff1a922025fc8284d8894b701033dbff46b with some additional changes from your similar PR #33 merged as 3143592b5ed17c9246000db220863a09f86c1970 on top.

Notably, 3143592b5ed17c9246000db220863a09f86c1970 removed `TestRelTimeMagnitudes`, which is where the tests that failed on `>` were located IIRC.

**Comment by dustin** [comment]
Interesting.  I do wish go had a better quickcheck.  I've found lots of these annoying edge case bugs with property tests.  I've got a few good ones at work, but it's so much harder to do it well in go than it is in haskell that you can get lost in the details.

**Comment by shurcooL** [comment]
> Yeah, I started using the non-mono one.  We don't have hard line length limitations at work, and go doesn't do very fancy alignment, so I gave it a shot to see how I'd like it.  It's not that bad, but it was a pretty big leap to program without a fixed width font.  I still used a fixed with font for everything else.

I see, cool. :) As far as I know, Rob Pike uses a non-mono font for coding too. I might give it a try for a longer period of time someday.

Either way, I do enjoy the Go fonts quite a bit (see my mini-review of it [here](https://twitter.com/shurcooL/status/900220414672154625), lol). In addition to my code editor, I also use it on my [personal site](https://dmitri.shuralyov.com/), and I managed to get it into [`maintserve`](https://godoc.org/golang.org/x/build/maintner/cmd/maintserve) as well, hehe.
